jbpm-workitems/jbpm-workitems-email/src/main/java/org/jbpm/process/workitem/email/TemplateManager.java

```diff
@@ -0,0 +1,201 @@
+            
+            if (directory.exists() && directory.isDirectory()) {
+                File[] foundTemplates = directory.listFiles((dir, name) -> { return name.endsWith(".html");});
+                
+                for (File templateFile : foundTemplates) {
+                    
+                    loadTemplate(templateFile);
+                }
+            }
+        }
+    }
+    
+    protected void loadTemplate(File templateFile) {
+        String templateId = templateFile.getName().substring(0, templateFile.getName().lastIndexOf("."));
+        
+        try (FileInputStream inputStream = new FileInputStream(templateFile)) {
+            this.stringLoader.putTemplate(templateId, read(inputStream));
+            
+            logger.info("Loaded template {} from file {}", templateId, templateFile);
+        } catch (Exception e) {
+            logger.warn("Exception while loading template from {} due to {}", templateFile, e.getMessage(), e);
+        }
+    }
+    
+    protected void removeTemplate(File templateFile) {
+        String templateId = templateFile.getName().substring(0, templateFile.getName().lastIndexOf("."));
+        stringLoader.removeTemplate(templateId);
+        logger.info("Removed template {} backed by file {}", templateId, templateFile);
+    }
+     
+    protected String read(InputStream input) {
+        String lineSeparator = System.getProperty("line.separator");
+
+        try (BufferedReader buffer = new BufferedReader(new InputStreamReader(input, Charset.forName("UTF-8")))) {
+            return buffer.lines().collect(Collectors.joining(lineSeparator));
+        } catch (Exception e) {
+            return null;
+        }
+    }
+    
+    public void close() {
+        if (this.watcher != null) {
+            this.watcher.stop();
+            this.watcher = null;
+        }
+    }
+    
+    private class TemplateDirectoryWatcher implements Runnable {
+     
+        private WatchService watcher;
+        private Path toWatch;
+        private AtomicBoolean active = new AtomicBoolean(true);     
+        
+        public TemplateDirectoryWatcher(String configFilePath) {
+            
+            this.toWatch = Paths.get(configFilePath);
+            
+            try {
+                this.watcher = toWatch.getFileSystem().newWatchService();
+                logger.debug("About to start watching " + toWatch.toString());
+                toWatch.register(watcher, ENTRY_MODIFY, ENTRY_CREATE, ENTRY_DELETE);
+            } catch (Exception e) {
+                logger.error("Error when setting up config file watcher :: " + e.getMessage(), e);
+                this.active.set(false);
+            }
+        }
+
+        public void stop() {
+            this.active.set(false);
+        }
+        
+        @Override
+        public void run() {
+            try{
+                while(active.get()) {
+                    WatchKey key = watcher.poll(5, TimeUnit.SECONDS);
```

Comment:
How about making the interval configurable via a system property?

Reply:
done